File: Ruby_Test/RbSample1/lib/dj/test/io/TestFile.rb
input_filename = "etc/SampleInput.txt"
puts "input_filename = #{input_filename}"

input_file_handle = open(input_filename)
puts "input_file_handle = #{input_file_handle}"
in_data = input_file_handle.read
puts "input data length = #{in_data.length}"

input_file_handle.close

puts "-" * 40
puts in_data
puts "-" * 40

output_filename = "etc/SampleOutput.txt"
puts "output_filename = #{output_filename}"

output_file_handle = open(output_filename, "w")
puts "output_file_handle = #{output_file_handle}"
output_file_handle.write(in_data)
puts "output data length = #{in_data.length}"

output_file_handle.close

File: Python_Test/PySample1/dj/test/io/TestFile.py
'''
Created on Mar 2, 2016

@author: dj
'''

with open('../../../etc/SampleInput.txt') as file1:
    print('file1 =', file1)

    for line in file1:
        print(line, end='')

print("-" * 40)

with open('../../../etc/SampleInput.txt', 'rb') as file2:
    print('file2 =', file2)

    for line in file2:
        print(line)

print("-" * 40)

file3 = open("../../../etc/SampleInput.txt")

print(file3.readline(), end="")
print("-" * 20)
print(file3.read(), end="")

print("-" * 20)

file3.seek(0)

print(file3.readline(6))
print("-" * 20)
print(file3.read(10))

file3.close()

print("-" * 40)

fileA = open("../../../etc/SampleOuput.txt", "w")

print("fileA.tell() =", fileA.tell())
fileA.write("Apple is not an orange.\n")
print("fileA.tell() =", fileA.tell())

fileA.close()

fileB = open("../../../etc/SampleOuput.txt", "a")

print("fileB.tell() =", fileB.tell())
fileB.write("Cat is not a dog.\n")
print("fileB.tell() =", fileB.tell())

fileB.close()

print("-" * 40)

file5 = open("../../../etc/SampleOuput.txt", "rb")

print(file5.read())

file5.close()

print("-" * 40)

if __name__ == '__main__':
    pass
